### PR TITLE
Fix race condition on removing aged deleted segments

### PR DESCRIPTION
**Pull request by jackjlli**
This PR fixes the race condition that multiple controllers trying to remove the segments for the same table.

Unit tests added.

### CODE REVIEW

**Comment by Jackie-Jiang** [comment]
Does it run into exception or just logged some warning? I'd suggest not introducing the lead controller concept into this module, but simply handle the case of file deleted during the checks because lead controller change is not guaranteed to be visible to all controllers at the same time.

**Comment by jackjlli** [comment]
> Does it run into exception or just logged some warning? I'd suggest not introducing the lead controller concept into this module, but simply handle the case of file deleted during the checks because lead controller change is not guaranteed to be visible to all controllers at the same time.

Yes, we ran into some exception that multiple controllers were trying to delete for the same table in Prod. 
There is a 5-min delay config in lead controller resource, which means there would be lead controller changes only when there is some discrepancy between IS and EV for 5 mins.
I think in this case it should be fine despite the lead controller change hasn't reached the controllers, as we just need a dedicated controller to clean up the aged segments. Even if one controller is permanently removed out of the cluster, it's okay to have the aged segments stay for another 5 mins and then another new controller got assigned to take the segment deletion. 

I'm fine with capturing the exception within a try-catch block and proceed with the rest of aged segment deletion for the rest of the tables.

**Review by Jackie-Jiang — APPROVED** [review]

File: pinot-common/src/main/java/org/apache/pinot/common/utils/URIUtils.java [hunk 1]
```diff
@@ -70,6 +70,14 @@ public static String getPath(String basePath, String... parts) {
     return stringJoiner.toString();
   }
 
+  /**
+   * Returns the last part for the given path split by the file separator.
+   * If the file separator is not found, returns the whole path as the last part.
+   */
+  public static String getLastPart(String path) {
```
Comment (PrachiKhobragade) [inline pinot-common/src/main/java/org/apache/pinot/common/utils/URIUtils.java:77]: If used at other places, the URI might have query parameters. Should those be removed? Or add a test case for "?..="

Comment (jackjlli) [reply pinot-common/src/main/java/org/apache/pinot/common/utils/URIUtils.java:77]: Good catch! Added a few more unit tests.

**Pull request merged**